flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/state/JdbcAutoScalerStateStore.java

```diff
@@ -0,0 +1,286 @@
+                .map(JdbcAutoScalerStateStore::deserializeParallelismOverrides)
+                .orElse(new HashMap<>());
+    }
+
+    @Override
+    public void removeParallelismOverrides(Context jobContext) {
+        jdbcStateStore.removeSerializedState(getSerializeKey(jobContext), PARALLELISM_OVERRIDES);
+    }
+
+    @Override
+    public void clearAll(Context jobContext) {
+        jdbcStateStore.clearAll(getSerializeKey(jobContext));
+    }
+
+    @Override
+    public void flush(Context jobContext) throws Exception {
+        jdbcStateStore.flush(getSerializeKey(jobContext));
+    }
+
+    @Override
+    public void removeInfoFromCache(KEY jobKey) {
+        jdbcStateStore.removeInfoFromCache(getSerializeKey(jobKey));
+    }
+
+    private String getSerializeKey(Context jobContext) {
+        return getSerializeKey(jobContext.getJobKey());
+    }
+
+    private String getSerializeKey(KEY jobKey) {
+        return jobKey.toString();
+    }
+
+    // The serialization and deserialization are similar to KubernetesAutoScalerStateStore
+    protected static String serializeScalingHistory(
+            Map<JobVertexID, SortedMap<Instant, ScalingSummary>> scalingHistory) throws Exception {
+        return compress(YAML_MAPPER.writeValueAsString(scalingHistory));
+    }
+
+    private static Map<JobVertexID, SortedMap<Instant, ScalingSummary>> deserializeScalingHistory(
+            String scalingHistory) throws JacksonException {
+        return YAML_MAPPER.readValue(decompress(scalingHistory), new TypeReference<>() {});
+    }
+
+    protected static String serializeScalingTracking(ScalingTracking scalingTracking)
+            throws Exception {
+        return compress(YAML_MAPPER.writeValueAsString(scalingTracking));
+    }
+
+    private static ScalingTracking deserializeScalingTracking(String scalingTracking)
+            throws JacksonException {
+        return YAML_MAPPER.readValue(decompress(scalingTracking), new TypeReference<>() {});
+    }
+
+    @VisibleForTesting
+    protected static String serializeEvaluatedMetrics(
+            SortedMap<Instant, CollectedMetrics> evaluatedMetrics) throws Exception {
+        return compress(YAML_MAPPER.writeValueAsString(evaluatedMetrics));
+    }
+
+    private static SortedMap<Instant, CollectedMetrics> deserializeEvaluatedMetrics(
+            String evaluatedMetrics) throws JacksonException {
+        return YAML_MAPPER.readValue(decompress(evaluatedMetrics), new TypeReference<>() {});
+    }
+
+    private static String serializeParallelismOverrides(Map<String, String> overrides) {
+        return ConfigurationUtils.convertValue(overrides, String.class);
+    }
+
+    private static Map<String, String> deserializeParallelismOverrides(String overrides) {
+        return ConfigurationUtils.convertValue(overrides, Map.class);
+    }
+
+    private static String compress(String original) throws IOException {
```

Comment:
Is there really a reason to compress the data in a DB? For configmaps this was done due to the very constrained size. But compressing makes it much harder to read for users

Reply:
No any strong reason, it refers the `KubernetesAutoScalerStateStore`.

I removed the the compression related logic for JdbcAutoScalerStateStore in the latest hotfix commit.